src/Helpers/SerializationHelper.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sammyjo20\LaravelHaystack\Helpers;
+
+use Illuminate\Database\PostgresConnection;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Str;
+
+class SerializationHelper
+{
+    /**
+     * Serialize the given value.
+     *
+     * @param  mixed  $value
+     * @return string
+     */
+    public static function serialize(mixed $value) : string
+    {
+        $serialized = serialize($value);
+
+        return DB::connection() instanceof PostgresConnection
```

Comment:
Should we be using the connection that has been specified on the model instead of the global DB connection? From Laravel's PR:

https://github.com/laravel/framework/pull/36081/files#diff-c46d3e14c232058da1ea59d469b8abc9866c1ced2104eece667c4077b54a5e22R304-R306

Reply:
Yes, I think you are right. This solution only covers the default scenario. I will look into it and make fixes accordingly.